central/processlisteningonport/datastore/datastore_impl.go

```diff
@@ -105,6 +107,40 @@ func (ds *datastoreImpl) AddProcessListeningOnPort(
 		}
 
 		indicatorsMap[key] = val
+
+		indicatorIds = append(indicatorIds, val.GetId())
+	}
+
+	if len(indicatorIds) > 0 {
+		// If no corresponding processes found, we can't verify if the PLOP
+		// object is closing an existing one. Collect existingPLOPMap only if
+		// there are some matching indicators.
+
+		// XXX: This has to be done in a single join query fetching both
+		// ProcessIndicator and needed bits from PLOP.
+		existingPLOP, err := ds.storage.GetByQuery(ctx, search.NewQueryBuilder().
+			AddStrings(search.ProcessID, indicatorIds...).
```

Comment:
Above we are using DisjunctionQuery. Here we are using an array of indicatorIds. We could use DisjuctionQuery here or we could use an array of indicatorIds above were we use DisjunctionQuery. What are the advantages and disadvantages of the methods? Why use these two methods? I don't know that one is better than the other.